assets/js/lib/model/checks.js

```diff
@@ -0,0 +1,28 @@
+import { pipe, some, castArray, filter } from 'lodash/fp';
+import { TARGET_CLUSTER } from '.';
+
+const hasMetadata = ({ metadata }) => !!metadata;
+const hasClusterTypeMetadata = ({ metadata }) =>
+  hasMetadata({ metadata }) && !!metadata.cluster_type;
+
+const isCheckForTarget =
+  (targetType) =>
+  ({ metadata }) =>
+    !hasMetadata({ metadata }) || metadata.target_type === targetType;
+
+export const hasChecksForTarget = (catalog, targetType) =>
+  some(isCheckForTarget(targetType))(catalog);
+
+export const hasChecksForClusterType = (catalog, clusterType) =>
+  pipe(
+    filter(isCheckForTarget(TARGET_CLUSTER)),
+    some(
+      ({ metadata }) =>
+        !hasMetadata({ metadata }) ||
```

Comment:
Isn't this check already done in the `filter()` step, and also later in `hasClusterTypeMetadata()`?

Reply:
yes, but it is a different check.
`hasClusterTypeMetadata` checks whether there are metadata AND there is a `cluster_type` in it.

`!hasMetadata({ metadata }) ||` usage hear means that if there are not metadata we consider the check applicable for the given `clusterType`

Reply:
nevermind, that seems to be safely removable 1408a32d91f6ba4b83ad3ccc95cabb38f4f7588f